src/lib/client.ts

```diff
@@ -168,6 +163,27 @@ const confirmedServiceMap: BACnetEventsMap = {
 	[beC.CONFIRMED_PRIVATE_TRANSFER]: 'privateTransfer',
 }
 
+class Deferred<T> {
+	resolve: (value: T) => void
+	reject: (err: Error) => void
+
+	promise: Promise<T>
+
+	constructor() {
+		this.promise = new Promise((resolve, reject) => {
+			this.resolve = resolve
+			this.reject = reject
+		})
+	}
+}
```

Comment:
IMO we could move all invokeStore logic into a separated class so we cloud remove some internal client methods there. WDYT?

Reply:
Of course also into a separeted file

Reply:
Haha, you read my mind. I had come to the same conclusion. Will do!

Reply:
Done in https://github.com/bacnet-js/client/pull/52/commits/f900d501850d6344dbec908fd11e45d0077fe094